### ISSUE TITLE
Bugged audio slider

### ISSUE THREAD

**Issue by Eken-beep** [scorpion-26/gBar#13]
Audio slider is basically useless for me due to the fact that when dragging the volume up or down the volume only starts changing in the rightmost couple pixels of the slider. The rest of the slider doesn't do anything but i can change the volume correctly in the rightmost part of the slider, goes from 100-0 in a very small area.

**Steps to Reproduce**
Open gBar through terminal
Use audioslider.

**Expected behavior**
The volume is at 90% when 90% of the slider is slid and 10% when 10% is slid.

**Screenshots/Error logs**
Not much to go on here...

**Information about your system and gBar**
 - OS: [NixOS]
 - Desktop environment [Hyprland]

**Comment by scorpion-26** [comment]
A few questions that may come in handy:
- Pulseaudio or Pipewire?
- Have you modified the CSS?

Can you do the following?
- Modify the ```border-style``` of the css class ```trough``` from ```none``` to ```solid``` (Refer to https://github.com/scorpion-26/gBar#i-want-to-customize-the-colors on a guide on how to modify the CSS)
- It should look like this: 
![Image](https://user-images.githubusercontent.com/58082714/226651054-a9808c97-9a49-476f-a10a-46f594cd82fe.png)
 The white rectangle above the audio slider denotes the area where you can drag it properly and it should *always* be attached to the leftmost edge of the slider (Like in the image). Can you reproduce the bug and see if there is anything odd happening with that white rectangle?

**Comment by scorpion-26** [comment]
Sorry, I meant ```slider``` instead of ```trough```

**Comment by Eken-beep** [comment]
![20230321_16h53m05s_grim](https://user-images.githubusercontent.com/84442052/226665434-783fdad6-a518-4654-a481-e20204be505c.png)
It follows the edge of the slider correctly but audio isn't modified until the last bit.

**Comment by scorpion-26** [comment]
So if I understand correctly, the slider works like it should, but the audio volume is not set properly.
If that's the case, I just added a new commit which adds some more logs (2fa23a2d54f2f4f73814b52cacb09b9dd98f3369). Can you check that out and post the logs when sliding the slider?

**Comment by scorpion-26** [comment]
> The issue seems to be with either pulse, pipewire or pamixer as setting the volume to 10 changes nothing. The bars audio slider works fine and mirrors pamixers volume. When i set the volume to 3 it changes drastically from 10.

That sounds very much like a bug in pipewire or the way your Pipewire is setup. I don't know how this correlates on NixOS, but on Arch you need a seperate package for PulseAudio APIs (Called pipewire-pulse). For things to work smoothly between PulseAudio clients (Like gBar and pamixer), you need to enable and start the pipewire-pulse systemctl service. You also *need* to make sure, that pulseaudio.service is not running anymore, since that collides with pipewire-pulse. I have a strong suspicion, that the PulseAudio and PipeWire are active at the same time, since the PulseAudio APIs don't seem to affect Pipewire. That suspicion could be confirmed if pavucontrol also doesn't work.
To summarize: pipewire.service and pipewire-pulse.service need to be running whereas pulseaudio.service shouldn't. If that's not the culprit, I'm thinking of adding a config variable to modify the commands that will be called to set the volume (Currently hard-coded to ```pamixer --set-volume xx```)

**Comment by Eken-beep** [comment]
I currently have pulse set to disabled and pipewire enabled with support for alsa and pulse (which i guess enables that service you talked about). A config option for max and min volume would be fantastic and solve my issue.

**Comment by scorpion-26** [comment]
Okay, implementing that

**Comment by scorpion-26** [comment]
Added a Min/Max volume config option(Called ```AudioMinVolume``` and ```AudioMaxVolume``` in the config). This essentially remaps the range. Values above 100% are allowed. Let me know if this can fix the issue.

**Comment by Eken-beep** [comment]
Yes that does the trick, while it isnt too smooth using only 7 levels on the entire slider it works perfectly fine.
